Ryujinx.Graphics.Gpu/Shader/GpuAccessorBase.cs

```diff
@@ -11,74 +13,139 @@ namespace Ryujinx.Graphics.Gpu.Shader
     class GpuAccessorBase
     {
         private readonly GpuContext _context;
+        private readonly ResourceCounts _resourceCounts;
+        private readonly int _stageIndex;
 
         /// <summary>
         /// Creates a new GPU accessor.
         /// </summary>
         /// <param name="context">GPU context</param>
-        public GpuAccessorBase(GpuContext context)
+        public GpuAccessorBase(GpuContext context, ResourceCounts resourceCounts, int stageIndex)
         {
             _context = context;
+            _resourceCounts = resourceCounts;
+            _stageIndex = stageIndex;
         }
 
-        /// <summary>
-        /// Queries host about the presence of the FrontFacing built-in variable bug.
-        /// </summary>
-        /// <returns>True if the bug is present on the host device used, false otherwise</returns>
+        /// <inheritdoc/>
+        public int QueryBindingConstantBuffer(int index)
+        {
+            if (_context.Capabilities.Api == TargetApi.Vulkan)
+            {
+                return 1 + GetBindingFromIndex(index, _context.Capabilities.MaximumUniformBuffersPerStage, "Uniform buffer");
```

Comment:
Perhaps add a comment as to why the + 1 is significant.
```suggestion
                return GetBindingFromIndex(index, _context.Capabilities.MaximumUniformBuffersPerStage, "Uniform buffer") + 1;
```

Reply:
The binding "0" is reserved to the used by the uniform buffer with the render scales and other emulator data that the shader needs to access.